src/components/course/CoursePage.jsx

```diff
@@ -68,43 +72,71 @@ const CoursePage = () => {
   // the URL query parameters and then remove it to keep the URLs clean.
   const algoliaSearchParams = useExtractAndRemoveSearchParamsFromURL();
 
+  const courseService = useMemo(() => new CourseService({
```

Comment:
Abstracts the instantiation of `CourseService` into the `CoursePage` component so it can shared between React hooks (i.e., `useAllCourseData ` and `useUserSubsidyApplicableToCourse`).